tests/tests_fabric/plugins/environments/test_slurm.py

```diff
@@ -129,6 +130,15 @@ def test_detect():
     with mock.patch.dict(os.environ, {"SLURM_JOB_NAME": "interactive"}):
         assert not SLURMEnvironment.detect()
 
+    with mock.patch.dict(os.environ, {"SLURM_NTASKS": "2"}), mock.patch(
```

Comment:
FWIW, I found a single env var to be responsible for work/fail

unsetting `SLURM_JOB_NAME` would fix things, so I wonder if perhaps it should be tested instead?

Reply:
Yes you got it to work this way but it is not the intended use. What you did with this environment variable is to tell Lightning that you are expected to be in interactive mode: https://lightning.ai/docs/pytorch/stable/clouds/cluster_advanced.html#interactive-mode 
This use case is when a user schedules a machine in SLURM and then connects to it to run the scripts manually. This is referred to often as interactive mode. Without explicitly telling Lightning that, it wouldn't be able to distinguish that from job submission mode. The difference is that in job submission mode, SLURM would be the one launching the processes, whereas in interactive mode the expectation is that Lightning will be responsible for launching the jobs. 

Your use case is different (given my understanding is right), you want to launch your processes yourself with torchrun. Whether you are in job submission mode or interactive doesn't matter here. All you care is that Lightning will recognize the torchrun environment variables and prioritize them over the settings in the slurm environment variables. So in this PR, I'm enabling that. If you want, you could install this branch and test it out :))

Reply:
ok, I will test it today and report back.

Thank you for working on this, Adrian.